### ISSUE TITLE
Old uploaded files aren't re-indexed when I enable them

### ISSUE THREAD

**Issue by Korayem** [langgenius/dify#1567]
### Dify version

0.3.30

### Cloud or Self Hosted

Cloud

### Steps to reproduce

1. Upload a file and wait for it to index
2. Wait for a couple of months
3. The file will be automtically disabled by the system and the index count will be Zero
4. Try to enable the file again

### ✔️ Expected Behavior

The file should re-index its content

### ❌ Actual Behavior

The file count is still Zero

Here I pressed "Enable" on all old files that became stale and got disabled, but the index count remained Zero
![image](https://github.com/langgenius/dify/assets/198332/7a230fc7-a92e-4fc4-bd25-a7a4d7714a84)

The only workaround is to upload the files again

**Comment by JohnJyong** [comment]
the retrieval count is not the index count， retrieval count mean how many times the document has been retrievered. You can check the documents' status at Retrieval Testing
![image](https://github.com/langgenius/dify/assets/76649700/f7ec2efb-8468-4056-914d-a069ccdbb475)

**Comment by Korayem** [comment]
My bad for the misunderstanding

I uploaded those documents more than 6 months ago. When I tried to test my existing chatbot tied to those Knowledge Docs, it didnt hit any doc, because no citations exist. So I went to the Knowledge and found all files got disabled. So I thought it must be an automated thing to save on space for idle accounts. So I enabled all files. Waited a bit and went back to ask the chatbot but still zero hits.

When I tried today, however, everything works extremely well!! I would say even better than the same GPT I have on chatgpt with the same files. The rerank using cohere is making a lot of difference apparently!

Cheers 🍻
